lib/data/external/GCP/GcpApis/copyObject.js

```diff
@@ -0,0 +1,107 @@
+const async = require('async');
+const request = require('request');
+const { errors } = require('arsenal');
+
+const { logger, JsonError, checkSourceString } = require('../GcpUtils');
+const { logHelper } = require('../../utils');
+
+/**
+ * copyObject - minimum required functionality to perform object copy
+ * for GCP Backend
+ * @param {object} params - update metadata params
+ * @param {string} params.Bucket - bucket name
+ * @param {string} params.Key - object key
+ * @param {string} param.CopySource - source object
+ * @param {function} callback - callback function to call with the copy object
+ * result
+ * @return {undefined}
+ */
+function copyObject(params, callback) {
+    const { CopySource } = params;
+    if (!checkSourceString(CopySource)) {
+        if (typeof CopySource === 'string') {
+            return callback(errors.InvalidArgument);
+        }
+        return callback(errors.MissingParameter);
+    }
+    const sourceArray = CopySource.split('/');
+    const sourceBucket = sourceArray[0] ? sourceArray[0] : sourceArray[1];
```

Comment:
If you just get rid of the first element of `sourceArray` if empty, you can probably simplify the following code by not having to deal with `objectStart`.